src/ini.pest

```diff
@@ -1,13 +1,17 @@
-WHITESPACE = _{ " " }
+ini = { SOI ~ (section | pair | comment | blank_line)* ~ ws? ~ EOI }
+ws = { (" " | "\t")* }
+
+// section
+section = { ws ~ "[" ~ name ~ "]" ~ ws ~ newline? }
 char = { ASCII_ALPHANUMERIC | "." | "_" | "/" }
+newline = { NEWLINE }
+
+// key-value pair
+pair = { ws ~ name ~ ws ~ equals ~ ws ~ value ~ ws ~ newline? }
+equals = { "=" }
 name = @{ char+ }
 value = @{ char* }
 
-section = { "[" ~ name ~ "]" }
-property = { name ~ "=" ~ value }
-
-file = {
-	SOI ~
-	((section | property)? ~ NEWLINE)* ~
-	EOI
-}
+comment = { ws ~ delimiter ~ value? ~ newline? }
+delimiter = { "#" | ";" }
```

Comment:
I think I meant to name this `indicator`?

Reply:
Good question - we should introduce the term "reserved characters", or like the ECMAScript spec introduces: *lexical elements*, like the section, a pair and a comment. Where a section uses the *punctuators* section-key-delimiters (`[]`) and a pair the assignment-operator. A comment is started by the *CommentMarker* and delimited by the `NEWLINE`.

Reply:
1. I don't like calling the section name a section key, because keys are commonly used keys in key-value pair or has map. If you want to have a discussion about this, we should have it on the specification repo.
1. Comment marker or indicator doesn't matter to me, but we need to add this to the specification.
1. I'm not sure a delimiter makes any sense for a comment, as a delimiter is intended to separate two of something. The newline is, however, a line delimter.

Reply:
No need to discuss that further - i am fine with section-name. I just wanted to add alternatives. Indicator is a good word. But we should also consider adding the term of reserved characters we have to. I mean indicators are reserved - at least at the beginning of the line?

Reply:
Comment indicators are reserved, but only in the context of the first non-whitespace character on a new line. We can't say they are 100% reserved and in the same breath tell people that it may or may not be supported inside a key's name and/or a pair's value in the future, which is essentially what we did in the specification. It's going to confuse people a bit.

Reply:
You're right.